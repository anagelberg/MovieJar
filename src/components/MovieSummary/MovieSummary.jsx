import './MovieSummary.scss';
import Rating from '../Rating/Rating';

function MovieSummary({ movie }) {
    return (
        <div className="summary__text">
            <div>
                <h6 className="summary__title">
                    {movie.title}
                </h6 >
                <h6 className="summary__title">
                    ({movie.year})
                </h6>
            </div>
            <Rating rating={movie.publicRating} />
            <p className="summary__description">{movie.description}</p>
            <p className="summary__runtime">{movie.runTime} minutes</p>
        </div>
    )
}

export default MovieSummary;