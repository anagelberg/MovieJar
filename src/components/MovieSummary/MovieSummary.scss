@use "../../styles/_mixins.scss" as *;
@use "../../styles/_variables.scss" as *;

.summary {
  &__rating {
    color: $primary;

    @include flex-center;
    gap: 0.5rem;
    vertical-align: middle;
    vertical-align: center;

    @include tablet {
      font-size: 12px;
    }
  }

  &__description {
    max-height: 30vh;
    overflow: auto;
    text-align: left;
    @include tablet {
      @include max-lines(12);
    }
  }

  &__title {
    font-size: 18px;
    @include tablet {
      @include max-lines(3);
      font-size: 12px;
    }
  }

  &__description,
  &__genres,
  &__vibe {
    @include tablet {
      font-size: 10px;
    }
  }

  &__runtime {
    @include tablet {
      font-size: 8px;
    }
  }

  &__icon {
    @include small-icon;
  }

  &__text {
    height: 100%;
    @include flex-center($justify-content: space-around);
    flex-direction: column;
    text-align: center;
  }
}
