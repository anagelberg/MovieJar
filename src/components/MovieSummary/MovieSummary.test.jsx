import MovieSummary from "./MovieSummary";
import { render, screen } from "@testing-library/react";

const testMovie = {
    title: "The Godfather",
    year: 1972,
    description: "The aging patriarch of an organized crime dynasty transfers control of his clandestine empire to his reluctant son.",
    runTime: 175,
    publicRating: 9.2
}

test("renders MovieSummary component", () => {
    render(<MovieSummary movie={testMovie} />);
    expect(screen.getByText(testMovie.title)).toBeInTheDocument();
    expect(screen.getByText(`(${testMovie.year})`)).toBeInTheDocument();
    expect(screen.getByText(testMovie.description)).toBeInTheDocument();
    expect(screen.getByText(`${testMovie.runTime} minutes`)).toBeInTheDocument();
});

test("MovieSummary component matches snapshot", () => {
    const { container } = render(<MovieSummary movie={testMovie} />);
    expect(container.firstChild).toMatchSnapshot();
});