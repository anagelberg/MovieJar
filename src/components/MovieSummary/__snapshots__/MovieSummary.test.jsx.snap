// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSummary component matches snapshot 1`] = `
<div
  class="summary__text"
>
  <div>
    <div
      class="summary__title-container"
    >
      <h4
        class="summary__title"
      >
        The Godfather
      </h4>
      <h4
        class="summary__title"
      >
        (
        1972
        )
      </h4>
    </div>
    <p
      class="rating"
    >
      9.2 / 5
      <img
        alt="star"
        class="rating__icon"
        src="star.svg"
      />
    </p>
  </div>
  <p
    class="summary__description"
  >
    The aging patriarch of an organized crime dynasty transfers control of his clandestine empire to his reluctant son.
  </p>
  <p
    class="summary__runtime"
  >
    175
     minutes
  </p>
</div>
`;
